runtime/gc_glue_java/EnvironmentDelegate.hpp

```diff
@@ -197,6 +200,17 @@ class MM_EnvironmentDelegate
 			/* calculate this BEFORE we (potentially) destroy the object */
 			_gcEnv.movedObjectHashCodeCache.originalHashCode = computeObjectAddressToHash((J9JavaVM *)_extensions->getOmrVM()->_language_vm, objectPtr);
 		}
+
+#if defined(J9VM_ENV_DATA64)
+		if (objectModel->isIndexable(objectPtr)) {
+			GC_ArrayObjectModel *indexableObjectModel = &_extensions->indexableObjectModel;
+			if (_vmThread->isIndexableDataAddrPresent && indexableObjectModel->isInlineContiguousArraylet((J9IndexableObject *)objectPtr)) {
```

Comment:
like for copy-forward case, it's better to check is offheap enabled (to avoid time waste for offheap disabled)